homeassistant/helpers/entity.py

```diff
@@ -317,9 +317,7 @@ def _async_write_ha_state(self):
         if not self.available:
             state = STATE_UNAVAILABLE
         else:
-            state = self.state
```

Comment:
I think we should keep this since `self.state` is a property. We don't need to calculate it twice if we keep this.

Reply:
Keeping it would broaden the type of `state` to everything `self.state` can be and require further workarounds/annoyances. But addressed using another temp variable.